pkg/trace/writer/trace.go

```diff
@@ -139,6 +145,49 @@ func (w *TraceWriter) Run() {
 	}
 }
 
+func (w *TraceWriter) runSync() {
+	defer close(w.stop)
+	defer close(w.flushChan)
+	for {
+		select {
+		case pkg := <-w.In:
+			w.addSpans(pkg)
+		case notify := <-w.flushChan:
+			w.drainAndFlush()
+			// Wait for encoding/compression to complete on each payload,
+			// and submission to senders
+			w.wg.Wait()
```

Comment:
Let's move this into `drainAndFlush` it is needed in all cases. And remove it from `Stop()` method.